components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/ConfigurationLoader.java

```diff
@@ -161,6 +160,49 @@ public static void loadInternalApis(String apiFilePath) {
         }
     }
 
+    private static void setHandlers(OMElement apiElement, InternalAPI api) {
+
+        List<InternalAPIHandler> handlerList = new ArrayList<>();
+        OMElement handlersOM = apiElement.getFirstChildWithName(new QName("handlers"));
+        if (handlersOM != null) {
+            Iterator<OMElement> handlers = handlersOM.getChildElements();
+
+            while (handlers.hasNext()) {
+                OMElement handlerOM = handlers.next();
+                if (handlerOM.getAttribute(new QName("name")) != null) {
+                    String handlerName = handlerOM.getAttributeValue(new QName("name"));
+                    if (handlerOM.getAttribute(new QName("class")) != null) {
+                        String handlerClass = handlerOM.getAttributeValue(new QName("class"));
+                        InternalAPIHandler handler = createHandler(handlerClass);
+                        handler.setName(handlerName);
+                        handlerList.add(handler);
+                    } else {
+                        handleException("Class not defined in one or more handlers");
```

Comment:
print a descriptive error message with the handler name

Reply:
Fixed